sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -319,15 +319,7 @@ case class PreprocessTableCreation(sparkSession: SparkSession) extends Rule[Logi
       conf.resolver)
 
     if (schema.nonEmpty && normalizedPartitionCols.length == schema.length) {
-      if (DDLUtils.isHiveTable(table)) {
```

Comment:
Can we check the commit history here? It seems we intentionally exclude hive table here.

Reply:
There doesn't seem to be any relevant information in the commit history message, It seems we did this on purpose from comment.

But in [HiveClientImpl.toHiveTable](https://github.com/apache/spark/blob/1ef5638177dcf06ebca4e9b0bc88401e0fce2ae8/sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala#L1069-L1072), we partitioned the partition cols and other cols, If all columns are partitioned columns, `hivetabl.getFields` will get an empty result, so Hive will throw an exception cols has at least one column

If Hive allows cols to inherit partitioned columns, we should not do `partition` in `toHiveTable`, if not, we should fail early, I'm sorry I'm not sure about that

Reply:
Sorry just got to this.
> If Hive allows cols to inherit partitioned columns, we should not do partition in toHiveTable, if not, we should fail early, I'm sorry I'm not sure about that
... what do we mean by "cols to inherit partitioned columns"?